Feature/se 49

bishop6232 opened the pull request:
closes SE2-WS23/documentation#49

![r_2880659_Vum14](https://github.com/SE2-WS23/Skillverse-Web/assets/104029083/19c1d84f-5c11-4ad4-b1ca-4f950896d458)

bishop6232 committed:
Add: basic article page implementation
doc: added docmentation to Article page

bishop6232 committed:
add: added the route for the article page

julix14 commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,48 @@
+			flexDirection: "column",
+			justifyContent: "center",
+			alignItems: "center",
+			height: "calc(100% - 60px)"
```
Why do you reduce the height by 60px?

bishop6232 replied:
I did that to leave a small space at the bottom which demarcate the container and the Next button.
you would have seen the effect if the container was coloured

julix14 replied:
Why didn't you use a margin for the bottom?

bishop6232 replied:
there is no specific reason I didn't use it. Margin-bottom can also achieve same purpose, I don't mind changing it

bishop6232 replied:
but I think the calc function is more suitable to achieve what I wanted

julix14 commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,48 @@
+			}}>
+
+
+			{/* box where the article is displayed */}
```
I think we can delete this comment here, because it is obvious that he Box holds the Article

bishop6232 replied:
done

julix14 commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,48 @@
+			variant="contained"
+			color="primary"
+			size='large'
+			sx={{position: "absolute",bottom: "1rem",right: "2rem"}} >
```
Maybe you can position the button on the same position as on the "linked-in-learning"-Page, so we have an uniform look

bishop6232 replied:
fixed that

julix14 commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,48 @@
+				<fieldset>
+					<legend >Article </legend>
+					<div style={{ padding: '5px' }}>
+						Article is display here
+						<p><a href="https://www.example.com" target="_blank" rel="noopener noreferrer">Visit Website</a></p>
+					</div>
```
Please adapt in a way, that we are able to pass the article dynamically into the page.
maybe you can use an demo object in the function that we can replace later with a correct article

bishop6232 replied:
I have fixed that, check if that's what you want

bishop6232 committed:
fix: i made the article page dynamic

julix14 reviewed (changes requested):
The code overall looks good, but please reformat your code with prettiert (default settings) I think you used "tabs" instead of "spaces"
Yours:
<img width="267" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/65715000/644281ac-5fb6-4753-bceb-6541ba3aa82f">


Formatted correctly:
<img width="489" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/65715000/7bc13b33-7c18-4a0a-9c05-97cd9afe7ed5">

julix14 commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,64 @@
+				}}>
+
+				<Box  sx={{height: "80%",width: "40%",display:"flex",justifyContent: "center",alignItems: "center",backgroundColor: "#D9D9D9", padding:"20px", overflowY:"auto" }}>
+					<fieldset>
```
Why did you choose the fieldset-Tag here?

bishop6232 replied:
I used the fieldset-Tag so I could group the content and use the legend-Tag to indicate the article title.

Chrissydrx replied:
I think I fieldset is not the best to display an article, but I also think, that the Mock-up uses a wrong representation

bishop6232 replied:
removed the feildset

bishop6232 commented:
what is the tab size used, 2 or 4?

bishop6232 committed:
fix: formatted the code to the required format

Chrissydrx commented on src/main.jsx:
```diff
@@ -47,6 +48,10 @@ const router = createBrowserRouter([
         element: <LinkedInLearning />,
       },
+      {
+        path: "Article",
```
```suggestion
        path: "article",
```

julix14 replied:
```suggestion
        path: "article",
```

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,64 @@
+ * Renders the Article page.
+ * @returns {JSX.Element} ArticlePage component
+ */
+function ArticlePage({article}) {
```
Please pass as props for consistency

bishop6232 replied:
changed

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
Please add prop types

bishop6232 replied:
done

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
Your component cannot handle a lot of text.
<img width="1728" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/37250473/61c558cc-8d43-4d69-9c5e-63cc65afe1e3">

bishop6232 replied:
fixed

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,64 @@
+	};
+	
+	
+	const articleDisplay = article || articleDemo; {/* if no article is passed then the demo article object above is displayed */ }
```
We should not ship a demo article, so please don't implement something like this. I think for now it is sufficient, if you just use the demoArticle. Also, for now we don't need to pass any props to the page, because we don't know how the content is passed to page (DB, ID etc)

bishop6232 replied:
if we didn't need to pass any props, should I remove the prop and use only the demoArticle for representation?

bishop6232 replied:
yea, removed it

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,64 @@
+				overflowY: "auto"
+				}}>
+
+				<Box  sx={{height: "80%",width: "40%",display:"flex",justifyContent: "center",alignItems: "center",backgroundColor: "#D9D9D9", padding:"20px", overflowY:"auto" }}>
```
width: 40% is not sufficient for a viewport width of 900.
<img width="942" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/37250473/45aa9c1f-917d-43c0-a2ce-f65f8d83999c">
I think you can use a higher width or breakpoints

bishop6232 replied:
I increased the width to 60% and it looks better for article display, but it different from what we have on the mock-up. 
check it and see how it looks.

Chrissydrx replied:
I still think, that this is a waste of place, if you use a small device. Maybe you consider to use a min-width. The Mock-up is (sadly) not our only point of truth, so sometimes we have to adapt something, so that is sufficient.

bishop6232 replied:
<img width="829" alt="Screenshot 2023-11-13 at 10 06 28" src="https://github.com/SE2-WS23/Skillverse-Web/assets/104029083/c96b4556-862d-450a-8557-b86e54acdcbc">


how it looks for small device like iPad

bishop6232 committed:
fix: the page can handle large text now.
i removed the demoArticle.
i also increased the width and removed the fieldset.

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -33,37 +19,37 @@ function ArticlePage({ article }) {
-            </div>
-          </fieldset>
+          <div
+            style={{
+              border: '1px solid black',
+              padding: '5px',
+              overflow: 'hidden',
+              overflowY: 'auto',
+            }}
```
Please use sx tag

bishop6232 replied:
done

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -33,37 +19,37 @@ function ArticlePage({ article }) {
+              overflow: 'hidden',
+              overflowY: 'auto',
+            }}
+          > Article content here
```
Article content here is displayed all the time and does not make sense

bishop6232 replied:
removed

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -78,5 +64,10 @@ function ArticlePage({ article }) {
     </PageLayout>
   );
 }
+ArticlePage.propTypes = {
+  title: PropTypes.string,
+  content: PropTypes.string,
+  link: PropTypes.string,
+};
```
Your props are not required, but you do not provide default values

bishop6232 replied:
fixed

Chrissydrx commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -33,37 +19,37 @@ function ArticlePage({ article }) {
+            flexDirection: 'column',
             justifyContent: 'center',
             alignItems: 'center',
             backgroundColor: '#D9D9D9',
```
Please use theme data for this color

bishop6232 replied:
the theme data has 4 colour, 3  of the colour is blue with different tone and the 4th one is contrastText which is just white

so am confused on the colour you want me to use, please explain better.

bishop6232 replied:
This is how it looks with  'light' from theme data.... is that okay for you?
<img width="1349" alt="Screenshot 2023-11-13 at 11 20 40" src="https://github.com/SE2-WS23/Skillverse-Web/assets/104029083/cec9560a-1b0d-4e22-9cc8-5b69a4da9f20">

j0world replied:
@Chrissydrx @julix14 Uzoma seems to have worked on all of his points. 
Is this done now?

julix14 replied:
No, he has to resolve the Code_Quality Issues

julix14 replied:
And his page throws many errors in the browser console
<img width="886" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/65715000/202456ef-db56-40f7-a9e5-981192eb8303">

bishop6232 committed:
fix: i used the the sx tag.
also made the prop required and provided fallback value.

bishop6232 committed:
Merge branch 'main' into feature/se-49

julix14 commented:
> SonarCloud Quality Gate failed.    [![Quality Gate failed](https://camo.githubusercontent.com/4ea51c1f64ee3746f631653a02ab678ca6a3efb5f5cb474402faed2e3dcf90b5/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f6661696c65642d313670782e706e67)](https://sonarcloud.io/dashboard?id=SE2-WS23_skillverse-web&pullRequest=13)
> 
> [![Bug](https://camo.githubusercontent.com/4c6102327f5a954f9c8acaf2e2714183157a9e41717b371b2cd585cf25057310/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6275672d313670782e706e67)](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=BUG) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=BUG) [![Vulnerability](https://camo.githubusercontent.com/3ba1ee49636ffc3427e38649a9f8a65ee392f28e8a662fcf96ce24cefbb520e9/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f76756c6e65726162696c6974792d313670782e706e67)](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=VULNERABILITY) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=VULNERABILITY) [![Security Hotspot](https://camo.githubusercontent.com/fb735cbe76f8d5e1679c76ce83b740ceb1eaf62de4f7bf88623dc9953261aff7/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f73656375726974795f686f7473706f742d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=SECURITY_HOTSPOT) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=SECURITY_HOTSPOT) [![Code Smell](https://camo.githubusercontent.com/8fe18b2dfb6f7d4e44582f281b29f617eb5ae07c248d2002ca586e91da219212/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f636f64655f736d656c6c2d313670782e706e67)](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=CODE_SMELL) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=CODE_SMELL) [2 Code Smells](https://sonarcloud.io/project/issues?id=SE2-WS23_skillverse-web&pullRequest=13&resolved=false&types=CODE_SMELL)
> 
> [![No Coverage information](https://camo.githubusercontent.com/86785eaa572e061f5ecb3f8bf1cbb7b8eca44925083f92d067805b4362cead0c/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f436f76657261676543686172742f4e6f436f766572616765496e666f2d313670782e706e67)](https://sonarcloud.io/component_measures?id=SE2-WS23_skillverse-web&pullRequest=13) No Coverage information [![0.0%](https://camo.githubusercontent.com/8047c63e1f9ed03f63001e1eadce4676bade3e0f83ec690a9c625287796248a6/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f4475706c69636174696f6e732f332d313670782e706e67)](https://sonarcloud.io/component_measures?id=SE2-WS23_skillverse-web&pullRequest=13&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=SE2-WS23_skillverse-web&pullRequest=13&metric=new_duplicated_lines_density&view=list)
> 
> ![idea](https://camo.githubusercontent.com/00150338e138e454e357462ed3d9cfceaf338ca50ccadcc1f3a75ad6a0e2aab2/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6c696768745f62756c622d313670782e706e67) Catch issues before they fail your Quality Gate with our IDE extension ![sonarlint](https://camo.githubusercontent.com/e4a8f808c72d84f231c1400ce7f2c77a09afe3eadb6bde9ad2c208a46735d320/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f736f6e61726c696e742d313670782e706e67) [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=sonarcloud-welcome)

Please fix the Issues from the CodeAnalysis
<img width="778" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/65715000/e1ca5f55-1518-429b-97b6-12d0e8601020">

bishop6232 committed:
removed unused import

julix14 commented on src/main.jsx:
```diff
@@ -48,16 +47,13 @@ const router = createBrowserRouter([
-        element: <AssessmentPage />,
-      },
     ],
   },
+  {
+    path: "/assessment",
+    element: <AssessmentPage />,
+    errorElement: <ErrorPage />,
+  },
```
Fix the router please and restore the trainingplan page

julix14 commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,80 @@
+            justifyContent: 'center',
+            alignItems: 'center',
+            backgroundColor: theme.palette.primary.light ,
+            padding: '20px',
```
Dont work with px values here.
We have defined padding values in our UI Guidelines. These are the Default MUI Values so you can use them here directly https://mui.com/system/spacing/
<img width="615" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/65715000/df4910df-97d3-4273-bf28-bc908b76c5a0">

julix14 commented on src/pages/Article/ArticlePage.jsx:
```diff
@@ -0,0 +1,80 @@
+          <Typography
+            sx={{
+              border: '1px solid black',
+              padding: '5px',
```
Spacing again

julix14 replied:
https://mui.com/system/spacing/

bishop6232 committed:
fixed the router
and also corrected the missing imports after the merge from main

julix14 reviewed (changes requested):
<img width="886" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/65715000/73683cff-b8a7-4648-976b-e597ad6a073a">

Please fix all your errors occuring.

bishop6232 committed:
fixed: corrected the html tag error on the console
and also chnaged how the article link and and title is displayed

bishop6232 committed:
Merge branch 'main' into feature/se-49

bishop6232 committed:
add: article routes into routes file

bishop6232 committed:
Merge branch 'main' into feature/se-49

bishop6232 committed:
Merge branch 'main' into feature/se-49

Chrissydrx commented:
Resolve Changes and create new PR
